### PR TITLE
Rudimentary testing framework

### PR DESCRIPTION
**Pull request by marcelocantos**
Changes proposed in this pull request:
- Implement a rudimentary testing framework

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: rel/value_set_native_func.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ type NativeFunction struct {
 
 // NewNativeFunction returns a new function.
 func NewNativeFunction(name string, fn func(Value) Value) Value {
-	return &NativeFunction{"⧼" + name + "⧽", fn}
+	return &NativeFunction{"⦑" + name + "⦒", fn}
```
Comment (marcelocantos) [inline rel/value_set_native_func.go:19]: The other fancy tokens don't render on Linux.

File: syntax/std_tst.go [hunk 2]
```diff
@@ -37,8 +39,48 @@ func stdTest() rel.Attr {
...
 			createTestCheckFuncAttr("true", func(v rel.Value) bool { return v.IsTrue() }),
 		),
-		rel.NewNativeFunctionAttr("suite", func(value rel.Value) rel.Value {
-			return nil
+		rel.NewNativeExprFunctionAttr("suite", func(expr rel.Expr, local rel.Scope) (rel.Value, error) {
+			switch expr := expr.(type) {
+			case rel.Value:
+				return rel.None, nil
+			case *rel.SetExpr:
+				errors := []error{}
```
Comment (joshcarp) [inline syntax/std_tst.go:47]: `var errors []error`

Comment (joshcarp) [inline syntax/std_tst.go:47]: `var errors []error`

File: rel/expr.go [hunk 3]
```diff
@@ -35,3 +35,16 @@ func (e ExprScanner) Scanner() parser.Scanner {
 func wrapContext(err error, expr Expr) error {
 	return fmt.Errorf("%s\n%s", err.Error(), expr.Scanner().Context(parser.DefaultLimit))
```
Comment (joshcarp) [inline rel/expr.go:36]: `errors.Wrap(err, expr.Scanner().Context(parser.DefaultLimit))`

File: syntax/std_tst.go (hunk 2 above)
Comment (joshcarp) [inline syntax/std_tst.go:47]: `var errors []error`
or using `errors.Wrap(err, message)` might make it easier than assigning to a slice then concatenating the errors

File: rel/expr.go (hunk 3 above)
Comment (marcelocantos) [reply rel/expr.go:36]: Too hard. Breaks some tests.

File: syntax/std_tst.go (hunk 2 above)
Comment (marcelocantos) [reply syntax/std_tst.go:47]: The clean solution is a well-formatted test output log. This will do for now.

File: rel/expr.go (hunk 3 above)
Comment (marcelocantos) [reply rel/expr.go:36]: #224

**Pull request merged**